section_faiss_vector_index.go

```diff
@@ -398,16 +378,22 @@ func (vo *vectorIndexOpaque) writeVectorIndexes(w *CountHashWriter) (offset uint
 			metric = faiss.MetricInnerProduct
 		}
 
-		// create an index, its always a flat for now, because each batch size
-		// won't have too many vectors (in order for >100K). todo: will need to revisit
-		// this logic - creating based on configured batch size in scorch.
-		index, err := faiss.IndexFactory(int(content.dim), "IDMap2,Flat", metric)
+		// create an index - currently its a IVF with 2 clusters and no compression
+		index, err := faiss.IndexFactory(int(content.dim), "IVF2,Flat", metric)
```

Comment:
This means there's a requirement that the index needs at least 2 vectors? So if a batch comes in with a single vector - this'd fail.

Reply:
So should probably use IVF1.